server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestFunction.java

```diff
@@ -303,4 +304,291 @@ void diagnose_defined_after() {
+          + "       01  {$*UNRELATED-STUFF}.                                 \n"
+          + "           05  {$*NUM}         PIC X(1234).                     \n"
+          + "       PROCEDURE DIVISION RETURNING {$UNRELATED-STUFF}.        \n"
+          + "       END FUNCTION FUNC1.|FUNC1}                                  \n"
+          + "       \n"
+          + "       IDENTIFICATION DIVISION.\n"
+          + "       PROGRAM-ID. EXAMPLE.\n"
+          + "       ENVIRONMENT DIVISION.\n"
+          + "       CONFIGURATION SECTION.\n"
+          + "       DATA DIVISION.\n"
+          + "       WORKING-STORAGE SECTION.\n"
+          + "       01 {$*so-data}.\n"
+          + "           05 {$*func1} pic 9.\n"
+          + "       PROCEDURE DIVISION.\n"
+          + "           display {$func1}.\n"
+          + "           display function {$$func1}.\n"
+          + "           display \"hello\".";
+
+  @Test
+  void test_non_declared_function_name_can_be_used_as_var_name() {
+    UseCaseEngine.runTest(
+        NON_DECLARED_FUNCTION_NAME_CAN_BE_USED_AS_VAR_NAME, ImmutableList.of(), ImmutableMap.of());
+  }
+
+  //
+  public static final String DECLARED_FUNCTION_NAME_CANT_BE_USED_AS_VAR_NAME =
+      "       {$$*IDENTIFICATION DIVISION.                             \n"
+          + "       FUNCTION-ID. FUNC1.                                  \n"
+          + "       DATA DIVISION.                                       \n"
+          + "       LINKAGE SECTION.                                     \n"
+          + "       01  {$*UNRELATED-STUFF}.                                 \n"
+          + "           05  {$*NUM}         PIC X(1234).                     \n"
+          + "       PROCEDURE DIVISION RETURNING {$UNRELATED-STUFF}.        \n"
+          + "       END FUNCTION FUNC1.|FUNC1}                                  \n"
+          + "       \n"
+          + "       IDENTIFICATION DIVISION.\n"
+          + "       PROGRAM-ID. EXAMPLE.\n"
+          + "       ENVIRONMENT DIVISION.\n"
+          + "       CONFIGURATION SECTION.\n"
+          + "       repository.\n"
+          + "           FUNCTION {$$func1}\n"
+          + "           FUNCTION ALL Intrinsic.\n"
+          + "       DATA DIVISION.\n"
+          + "       WORKING-STORAGE SECTION.\n"
+          + "       01 {$*so-data}.\n"
+          + "           05 {$*func1|1} pic 9.\n"
+          + "           05 {$*hex-of|2} pic 9.\n"
+          + "       PROCEDURE DIVISION.\n"
+          + "           display {$func1}.\n"
+          + "           display function {$$func1}.\n"
+          + "           display \"hello\".";
+
+  @Test
+  void test_declared_function_name_cant_be_used_as_var_name() {
+    UseCaseEngine.runTest(
+        DECLARED_FUNCTION_NAME_CANT_BE_USED_AS_VAR_NAME,
+        ImmutableList.of(),
+        ImmutableMap.of(
+            "1",
+            new Diagnostic(
+                new Range(),
+                "Variable name FUNC1 is not allowed",
```

Comment:
... because of conflicting function declaration.

Reply:
Yes.
Compiler gives below o/p
```
   000020                    05 func1 pic 9.                                                                 000000000 1C

 ==000020==> IGYDS1266-E The name "FUNC1" was used for an item that was not defined as a
                         data-name.  References to this name may be resolved incorrectly.
```

Reply:
Add it to the message, so that the user is told as well.

Reply:
done